### PR TITLE
Add sdk gossip handler metrics

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

adds metrics to `gossip` package

## How this works

adds metrics

## How this was tested

n/a

### CODE REVIEW

File: network/p2p/gossip/gossip.go [hunk 1]
```diff
@@ -34,20 +36,42 @@ func NewGossiper[T any, U GossipableAny[T]](
 	log logging.Logger,
 	set Set[U],
 	client *p2p.Client,
-) *Gossiper[T, U] {
-	return &Gossiper[T, U]{
+	metrics prometheus.Registerer,
```
Comment (joshua-kim) [inline network/p2p/gossip/gossip.go:39]: Making sure people are fine w/ this as our signature. Avalanchego's context exposes a `MultiGatherer` so we'd be expected to register a `prometheus.NewRegistry` into the VM's context to use this.

File: network/p2p/gossip/handler.go [hunk 2]
```diff
@@ -70,14 +99,15 @@ func (h Handler[T]) AppRequest(_ context.Context, _ ids.NodeID, _ time.Time, req
...
 
-		// check that this doesn't exceed our maximum configured response size
+		// check that this doesn't exceed our maximum configured target response
+		// size
+		gossipBytes = append(gossipBytes, bytes)
+
 		responseSize += len(bytes)
-		if responseSize > h.maxResponseSize {
+		if responseSize > h.targetResponseSize {
 			return false
```
Comment (anusha-ctrl) [inline network/p2p/gossip/handler.go:108]: are the targetResponseChanges meant to be included in this PR?

File: network/p2p/gossip/gossip.go [hunk 3]
```diff
@@ -139,4 +166,7 @@ func (g *Gossiper[T, U]) handleResponse(
 			continue
 		}
 	}
+
```
Comment (anusha-ctrl) [inline network/p2p/gossip/gossip.go:169]: would it be helpful to add time metrics here as well ?

File: network/p2p/gossip/handler.go (hunk 2 above)
Comment (joshua-kim) [reply network/p2p/gossip/handler.go:108]: from diff on https://github.com/ava-labs/avalanchego/pull/1995, should be gone now that it's merged

File: network/p2p/gossip/gossip.go (hunk 3 above)
Comment (joshua-kim) [reply network/p2p/gossip/gossip.go:169]: we could but I think these will be strongly correlated with either the bytes/n which is why i didn't add it

Comment (joshua-kim) [reply network/p2p/gossip/gossip.go:169]: we have some visibility w/ tracers but I think this is still nice to have. added.

**Review by anusha-ctrl — APPROVED** [review]

File: network/p2p/gossip/gossip.go [hunk 4]
```diff
@@ -34,20 +38,54 @@ func NewGossiper[T any, U GossipableAny[T]](
...
+			Name:      "gossip_received_bytes",
+			Help:      "amount of gossip received (bytes)",
+		}),
+		receivedTime: receivedTime,
 	}
+
+	errs := wrappers.Errs{}
+	errs.Add(
+		metrics.Register(g.receivedN),
+		metrics.Register(g.receivedBytes),
```
Comment (darioush) [inline network/p2p/gossip/gossip.go:75]: does `receivedTime` need to be registered?

File: network/p2p/gossip/handler.go [hunk 5]
```diff
@@ -23,21 +27,60 @@ var (
...
+			Name:      "gossip_sent_bytes",
+			Help:      "amount of gossip sent (bytes)",
+		}),
+		sentTime: sentTime,
 	}
+
+	errs := wrappers.Errs{}
+	errs.Add(
+		metrics.Register(h.sentN),
+		metrics.Register(h.sentBytes),
```
Comment (darioush) [inline network/p2p/gossip/handler.go:66]: same with `sentTime`?

**Review by darioush — APPROVED** [review]

File: network/p2p/gossip/gossip.go (hunk 4 above)
Comment (joshua-kim) [reply network/p2p/gossip/gossip.go:75]: It's registered internally in https://github.com/ava-labs/avalanchego/blob/3d1040dc26bea0985c339f0876e6dd338ac8e234/utils/metric/averager.go#L46

File: network/p2p/gossip/gossip.go (hunk 1 above)
Comment (StephenButtolph) [reply network/p2p/gossip/gossip.go:40]: Yeah this function should be taking in a `prometheus.Registerer` 👍.

**Pull request merged**